File: blackened/maven-servlet/src/main/java/org/sayner/sandbox/example/web/servlet/WelcomeServlet.java
package org.sayner.sandbox.example.web.servlet;

import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;
import java.io.IOException;
import java.io.PrintWriter;

@WebServlet("/welcome")
public class WelcomeServlet extends HttpServlet {
    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
            throws IOException {
        final HttpSession session = req.getSession();

        Integer visitCounter = (Integer) session.getAttribute("visitCounter");
        if (visitCounter == null) {
            visitCounter = 1;
        } else {
            visitCounter++;
        }
        session.setAttribute("visitCounter", visitCounter);

        final String username = req.getParameter("username");

        resp.setContentType("text/html");
        final PrintWriter printWriter = resp.getWriter();
        printWriter.write("Welcome for my servlet!" + "<br>");
        if (username == null) {
            printWriter.write("Hello, Anonymous" + "<br>");
        } else {
            printWriter.write("Hello, " + username + "<br>");
        }
        printWriter.write("Page was visited " + visitCounter + " times.");
        printWriter.close();
    }
}

File: blackened/jetty-spring/src/main/kotlin/org/sayner/sandbox/example/jetty/WelcomeServlet.kt
package org.sayner.sandbox.example.jetty

import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@WebServlet("/")
class WelcomeServlet : HttpServlet() {
    override fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
        val printWriter = resp.writer
        printWriter.write("Hello from Jetty")
        printWriter.close()
    }
}
